tree/treeplayer/inc/TTreeReader.h

```diff
@@ -221,10 +222,15 @@ class TTreeReader: public TObject {
    Iterator_t end() const { return Iterator_t(); }
 
 protected:
+   using NamedProxies_t = std::unordered_map<std::string, std::unique_ptr<ROOT::Internal::TNamedBranchProxy>>;
    void Initialize();
-   ROOT::Internal::TNamedBranchProxy* FindProxy(const char* branchname) const {
-      return (ROOT::Internal::TNamedBranchProxy*) fProxies.FindObject(branchname); }
-   TCollection* GetProxies() { return &fProxies; }
+   ROOT::Internal::TNamedBranchProxy* FindProxy(const char* branchname) const
+   {
+      const auto proxyIt = fProxies.find(branchname);
+      return fProxies.end() != proxyIt ? proxyIt->second.get() : nullptr;
+   }
+
+   void AddProxy(ROOT::Internal::TNamedBranchProxy *p) { fProxies[p->GetName()].reset(p); }
```

Comment:
Could we assert that no prior `TNamedBranchProxy` existed for `p->GetName()`?

Reply:
Hi @Axel-Naumann , sure. Thanks for the feedback.